src/test/cypress/integration/exam/ExamAssessment.spec.ts

```diff
@@ -0,0 +1,184 @@
+                assessmentDashboard.submitModelingAssessment().then((assessmentResponse) => {
+                    expect(assessmentResponse.response?.statusCode).to.equal(200);
+                });
+                cy.login(student, '/courses/' + course.id + '/exams/' + exam.id);
+                cy.contains('2 of 10 points').should('be.visible');
+            });
+        });
+
+        it('Assess a text exercise submission', () => {
+            const exerciseTitle = 'Cypress Text Exercise';
+            courseManagementRequests.createTextExercise({ exerciseGroup }, exerciseTitle);
+            courseManagementRequests.generateMissingIndividualExams(exam);
+            courseManagementRequests.prepareExerciseStartForExam(exam);
+            cy.login(student, '/courses/' + course.id + '/exams/' + exam.id);
+            examStartEnd.startExam();
+            cy.contains(exerciseTitle).click();
+            cy.get('#text-editor-tab').type(textSubmission.text);
+            cy.contains('Save').click();
+            cy.get('#exam-navigation-bar').find('.btn-danger').click();
+            examStartEnd.finishExam();
+            cy.login(tutor, '/course-management/' + course.id + '/exams');
+            cy.contains('Assessment Dashboard', { timeout: 60000 }).click();
+            assessmentDashboard.startAssessing();
+            assessmentDashboard.addNewFeedback(7, 'Good job');
+            assessmentDashboard.submitTextAssessment().then((assessmentResponse) => {
+                expect(assessmentResponse.response?.statusCode).to.equal(200);
+            });
+            cy.login(student, '/courses/' + course.id + '/exams/' + exam.id);
+            cy.get('.question-options').contains('7 of 10 points').should('be.visible');
+        });
+    });
+
+    describe('Exam Programming Exercise Assessment', () => {
+        const examEnd = (Cypress.env('isBamboo') ? GROUP_SYNCHRONIZATION : 0) + 115000;
+
+        beforeEach('Create Exam', () => {
+            prepareExam(dayjs().add(examEnd, 'milliseconds'));
+        });
+
+        it('Assess a programming exercise submission (MANUAL)', () => {
+            cy.login(admin);
+            courseManagementRequests.createProgrammingExercise({ exerciseGroup }).then((progRespone) => {
+                const programmingExercise = progRespone.body;
+                courseManagementRequests.generateMissingIndividualExams(exam);
+                courseManagementRequests.prepareExerciseStartForExam(exam);
+                cy.login(student, '/courses/' + course.id + '/exams/' + exam.id);
+                examStartEnd.startExam();
+                cy.contains(programmingExercise.title).should('be.visible').click();
+                makeSubmissionAndVerifyResults(partiallySuccessful, programmingExercise.packageName, () => {
+                    cy.get('#exam-navigation-bar').find('.btn-danger').click();
+                    examStartEnd.finishExam();
+                    cy.get('.alert').should('be.visible');
```

Comment:
Waiting for an alert might not be enough here. The alert could also contain a warning or error instead of a success message (which I assume is shown here)